src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -341,22 +344,15 @@ private function dehydrateValue(mixed $value, LivePropMetadata $propMetadata, ob
         }
 
         if (\is_array($value)) {
-            if ($propMetadata->collectionValueType() && Type::BUILTIN_TYPE_OBJECT === $propMetadata->collectionValueType()->getBuiltinType()) {
-                $collectionClass = $propMetadata->collectionValueType()->getClassName();
-                foreach ($value as $key => $objectItem) {
-                    if (!$objectItem instanceof $collectionClass) {
-                        throw new \LogicException(sprintf('The LiveProp "%s" on component "%s" is an array. We determined the array is full of %s objects, but at least on key had a different value of %s', $propMetadata->getName(), $component::class, $collectionClass, get_debug_type($objectItem)));
-                    }
-
-                    $value[$key] = $this->dehydrateObjectValue($objectItem, $collectionClass, $propMetadata->getFormat(), $component::class, sprintf('%s.%s', $propMetadata->getName(), $key));
+            foreach ($value as $key => $objectItem) {
+                $type = \gettype($objectItem);
+                if ('object' === $type) {
+                    $type = $objectItem::class;
                 }
-            }
 
-            if (!$this->isValueValidDehydratedValue($value)) {
-                $badKeys = $this->getNonScalarKeys($value, $propMetadata->getName());
-                $badKeysText = implode(', ', array_map(fn ($key) => sprintf('%s: %s', $key, $badKeys[$key]), array_keys($badKeys)));
+                $propMetadata = new LivePropMetadata($key, new LiveProp(true), $type, false, true, null);
```

Comment:
Any reason for this change and the changes above? We already have a `$propMetadata` variable - is it not correct for some reason?

Reply:
The goal here is for each item of the array to go through the dehydrate method. I did that to support array of DTOs

Reply:
That makes sense - but I think the old code already did that. The main difference between the old and new code just seems to be that the old called `->dehydrateObjectValue()` on each item and the new code calls `->dehydrateValue()`.

But there is an important detail in the original code: to determine the "type" of each item in the array, we should NOT use `$objectItem::class`. Instead, we need to read the "property info" off of the property. Otherwise, we may allow for a value to be dehydrated that won't be able to be rehydrated later. Example:

```
class Foo
{
    public array $dtos1 = [];

    /** @var Dto[] */
    public array $dtos2 = [];
}
```

```php
$foo = new Foo();
$foo->dtos1 = [new Dto()];
$foo->dtos2 = [new Dto()];
```

In this situation, the `dtos1` property is *invalid*: we will *not* be able to hydrate this. We know that `dtos1` holds an array, but an array of what? That is the purpose of the `$propMetadata->collectionValueType()->getClassName();`: it tells us what the "metadata" (e.g. phpdoc) tells us the "type" of each item in the collection is. If that's missing, we COULD still successfully dehydrate, but hydration would fail. So, instead, we fail during dehydration to warn the user early.